File: online-game-store-api/OnlineGameStore.DAL/Entities/Genre.cs
using MongoDB.Bson.Serialization.Attributes;
using OnlineGameStore.DAL.BsonSerializers;
using System;
using System.Collections.Generic;

namespace OnlineGameStore.DAL.Entities
{
    public class Genre : BaseEntity
    {
        [BsonElement("CategoryID")]
        [BsonSerializer(typeof(IntToGuidBsonSerializer))]
        public override Guid Id { get; set; }

        [BsonElement("CategoryName")]
        public string Name { get; set; }
        public Guid? ParentGenreId { get; set; }
        public virtual Genre ParentGenre { get; set; }
        public string Description { get; set; }
        public string Picture { get; set; }
        public virtual ICollection<Genre> ChildGenres { get; set; } = new List<Genre>();

        public virtual ICollection<GameGenre> Games { get; set; }
    }
}

File: online-game-store-front/src/app/classes/genre.ts
import { Guid } from "guid-typescript";

export class Genre {
  [prop: string]: any;
  constructor(
  public id : string = Guid.EMPTY,
  public name : string = '',
  public parentGenreId: string = '',
  public parentGenre: Genre | null = null,
  public children : Genre[] | null = []
  ) { }
}